system/core/editor.php

```diff
@@ -149,7 +149,16 @@ function loadEditor($html) {
 					this.isFocusSet = true;
 				}
 
-				// initiate the "image_upload" plugin
+                wym._iframe.contentDocument.addEventListener('paste', (event) => {
+                    const pasteContent = (event.clipboardData || window.clipboardData).getData('text');
+                    const textarea = document.createElement('TEXTAREA');
+                    textarea.value = pasteContent;
+                    wym.paste(textarea.value);
```

Comment:
I actually suspect this textarea can be removed and you can just do `wym.paste(pasteContent)`. It looks like the `getData('text')` will already give the plaintext version of the clipboard contents, so going through the `textarea.value` does not do anything special (that's just a string variable essentially).

Reply:
I suspect it as well, but couldn't test it so tried to replicate it as faithfully as possible for now.

Reply:
I'm pretty sure that if it works for any other formatted text (e.g. from libreoffice, or a browser), it will work from Word too. The existing "Paste from Word" is actually not really Word-specific, it's just named that way because that's where the biggest junk comes from, I guess.

Reply:
fixed